### DISCUSSION TITLE
r figure for capstone project - Delaney

### DISCUSSION THREAD

**Discussion by drlawson** [RobertsLab/resources#1198] [Q&A] [answered]
Hi all, Steven and I were brainstorming figure ideas to combine as much information from my capstone as I can into one (easy to understand) figure. However, I am not super well versed in R so I did what I could but I'm looking for suggestions and help! I can't really put female/male gonadal stage on the x or y axis because it is ordinal data not continuous so the graph ends up looking really weird. Also, I would like to have something show the difference between the 2 populations based on location in the figure that includes both, but having 2 shapes looks weird. Maybe another color palette for that population? How would I do something like that/what else could I do? (The image with width and heights up to 7 cm includes both populations, and the one with widths and heights up to 6 cm includes only one) Thanks all, looking forward to your help!! - sincerely, an R amateur (lol!)[

<img width="689" alt="single population ggplot" src="https://user-images.githubusercontent.com/73162507/116842905-0fe2dc00-ab93-11eb-97ab-14d02229b234.png">
<img width="698" alt="both populations ggplot" src="https://user-images.githubusercontent.com/73162507/116842898-05c0dd80-ab93-11eb-8c98-42178c0ade03.png">
<img width="687" alt="both pop  w: shapes" src="https://user-images.githubusercontent.com/73162507/116843201-16258800-ab94-11eb-967e-bf355ecc72ba.png">

[ggplot capstone farm population only.TXT](https://github.com/RobertsLab/resources/files/6412947/ggplot.capstone.farm.population.only.TXT)
[ggplot capstone both populations.TXT](https://github.com/RobertsLab/resources/files/6412946/ggplot.capstone.both.populations.TXT)
[ggplot capstone both populations w: shapes.TXT](https://github.com/RobertsLab/resources/files/6412956/ggplot.capstone.both.populations.w.shapes.TXT)

](url)

**Comment by sr320** [comment]
Can you do one where you have Female stage on the y axis and width on x?

And maybe one where Female stage on y axis, male stage on x axis and symbol size be height?

**Reply by drlawson** [reply]
<img width="687" alt="Screen Shot 2021-05-03 at 3 26 06 PM" src="https://user-images.githubusercontent.com/73162507/116941096-e3c36b80-ac23-11eb-86e4-3576b500d04a.png">
<img width="697" alt="Screen Shot 2021-05-03 at 3 28 34 PM" src="https://user-images.githubusercontent.com/73162507/116941272-3bfa6d80-ac24-11eb-873a-f6c900e17458.png">
[githubrequest1 copy.TXT](https://github.com/RobertsLab/resources/files/6418162/githubrequest1.copy.TXT)
[githubrequest2 copy.TXT](https://github.com/RobertsLab/resources/files/6418163/githubrequest2.copy.TXT)

**Reply by drlawson** [reply]
both of these include both populations, but I can redo with just one population if needed :)

**Reply by sr320** [reply]
For the second one, these are likely means? can you plot all data, possibly using `jitter`

**Reply by drlawson** [reply]
Would that be for height? And do I add that inside the aes()? If so, doesn't change anything (maybe they are hiding behind larger dots since there are only 9 possible combinations of male and female stage due to it being an ordinal scale)...

**Reply by sr320** [reply]
not height, the female / male plot you have. Yes you are probably correct they are simply hiding now.

**Reply by drlawson** [reply]
would jitter help in that case? not sure how to use it to make the other points visible. Let me know, thanks for the help :)

**Comment by sr320** [comment]
Also can you post the data?

**Reply by drlawson** [reply]
https://docs.google.com/spreadsheets/d/1yEDEfA5WaUk4pF2ZteG3aAZf_pfOcyxzU849nq4mfmE/edit?usp=sharing

this should work. included just the data needed for this figure to make it easy :)

**Reply by sr320** [reply]
What other data is there?

**Comment by sr320** [comment]
Have a look at this for working with GitHub https://washington.zoom.us/rec/play/HoYr33lGLfwGRFNkf-rJI4Pw6TDbUtyPK-EhgOhRR3-V_lZTBBhAQoi4B5N5Jp9OVVCiIPkt3OWu6e_D.wUiaKzo8GfzbbvrN?continueMode=true

**Reply by sr320** [reply]
repo- https://github.com/sr320/clam-development

**Comment by sr320** [comment]
how about 

![g](http://gannet.fish.washington.edu/seashell/snaps/clam-development_-_main_-_RStudio_2021-05-03_17-08-53.png)

**Reply by drlawson** [reply]
I like this figure :) However, I don't understand why there are some points higher than 3 or 2.75, etc. Could you help me understand this?

**Reply by sr320** [reply]
that is just the `jitter`

**Reply by drlawson** [reply]
<img width="789" alt="orange:red w border" src="https://user-images.githubusercontent.com/73162507/117779053-26241400-b1f3-11eb-963f-d306b7c50d36.png">

**Reply by drlawson** [reply]
Used that code to make this one, should be the same just changed the theme around. See you Thursday!

**Comment by sr320** [comment]
at Lab Meeting it was suggest stacked bar plots might also have value

**Reply by drlawson** [reply]
ok I will try to make one of those. should i be trying to have male and female grades on the x and y?

**Reply by drlawson** [reply]
or something involving height? sorry, I'm having trouble visualizing that plot

**Comment by laurahspencer** [comment, marked as answer]
Here's an example: 
![image](https://user-images.githubusercontent.com/17264765/117380377-694a5400-ae8e-11eb-89a7-4608d362556d.png)
![image](https://user-images.githubusercontent.com/17264765/117381752-a49a5200-ae91-11eb-9a7b-7ea511d94b23.png)
![image](https://user-images.githubusercontent.com/17264765/117381770-ab28c980-ae91-11eb-8348-02c3f54b5b37.png)

Here's my R code to make these stacked barplots - hope that helps, let me know if you have any questions! 
[Histology-barplots.zip](https://github.com/RobertsLab/resources/files/6438254/Histology-barplots.zip)

**Reply by drlawson** [reply]
WOW that is gorgeous! Do you think I'm on the right track with this one? I remember Steven mentioning a while back not to do proportions so I tried to do the values. I will need to add male as well of course

<img width="443" alt="Screen Shot 2021-05-07 at 2 52 11 PM" src="https://user-images.githubusercontent.com/73162507/117512106-cfdd7980-af43-11eb-8294-8190bbc68acb.png">

**Reply by laurahspencer** [reply]
I think this shows differences between the spring dates and the summer date - you might want to make the sex ratio + male gamete stage figures too, to see if anything jumps out. But- for what it's worth I did still like the 4 scatter plots that steven made, since they show the relationship between sperm + egg stage for the same individual.

**Comment by drlawson** [comment]
<img width="716" alt="size prob  oranges" src="https://user-images.githubusercontent.com/73162507/118209872-54d20280-b41e-11eb-9019-70f7881bbba9.png">

How does this look for the size by continuity graph Steven? I will of course explain what continuity means in the figure description. It skips 4.00-4.50 because there was no data there, is that ok? R^2 = 0.298 (also should the y axis be in percent or out of 1? I have both versions because I wasn't sure)

**Reply by drlawson** [reply]
whoops I forgot to switch the ranges from 3.00-3.49 instead of 3.00-3.50. quick and easy fix :)

**Comment by drlawson** [comment]
<img width="825" alt="stage prob  oranges" src="https://user-images.githubusercontent.com/73162507/118210036-abd7d780-b41e-11eb-8069-6cd608f26f51.png">
also this graph - dashed is grade 3, dotted is grade 2, and plain is stage 1 (just for female) - as predicted grade 3 stayed pretty constant across sizes and stage 1 decreased strongly. The line for stage 2 seems to be impacted by the 7.00-7.50 bin (since we are doing proportions there was only one individual in that bin but it had a grade of 2 (100%) which seems to throw it off). I might try doing bins of 1 cm instead of 0.5 cm to see how that turns out. What do you think?

**Reply by sr320** [reply]
Can you add the number of samples for each size class (n)?

**Reply by sr320** [reply]
Better yet just have axis not be percentage.
